### PR TITLE
[BugFix] Map/struct type persist

### PR DESCRIPTION
**Pull request by zombee0**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

map/struct type persist to support create external table with map/struct column  and the type can be recovered when fe restart

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 20 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java [hunk 1]
```diff
@@ -269,6 +273,11 @@ public class GsonUtils {
             .registerTypeHierarchyAdapter(Table.class, new GuavaTableAdapter())
             .registerTypeHierarchyAdapter(Multimap.class, new GuavaMultimapAdapter())
             .registerTypeAdapterFactory(new ProcessHookTypeAdapterFactory())
+            // specified childcalss DeSerializer must be ahead of fatherclass,
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java:276]: Is it means that subtype defined in COLUMN_TYPE_ADAPTER_FACTORY should be declared ahead of COLUMN_TYPE_ADAPTER_FACTORY?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/persist/gson/GsonUtils.java:276]: yes

File: fe/fe-core/src/test/java/com/starrocks/catalog/TypeTest.java [hunk 2]
```diff
@@ -193,4 +195,40 @@ public void testCanonicalName() {
...
+        Assert.assertEquals("MAP<INT,STRUCT<c1:int(11), cc1:varchar(1048576)>>", deType.toString());
+        // test initialed selectedField by ctor in deserializer.
+        deType.setSelectedField(0, false);
+    }
+
+    @Test
+    public void testStructSerialAndDeser() {
+        // "struct<struct_test:int,c1:struct<c1:int,cc1:string>>"
+        StructType c1 = new StructType(Lists.newArrayList(
+                new StructField("c1", ScalarType.createType(PrimitiveType.INT)),
```
Comment (Smith-Cruise) [inline fe/fe-core/src/test/java/com/starrocks/catalog/TypeTest.java:220]: UT is not covered for the struct subfield's comment.

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 20 / 20 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MapType.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/StructType.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|5|5|100.00%|[]|

**Review by Smith-Cruise — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**